tools/mage/end-to-end.go

```diff
@@ -0,0 +1,154 @@
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+	"github.com/magefile/mage/target"
+)
+
+// EndToEnd namespace
+type EndToEnd mg.Namespace
+
+var (
+	testDatabaseName = "ttn_lorawan_test"
+	databaseURI      = "postgresql://root@localhost:26257/" + testDatabaseName + "?sslmode=disable"
+)
+
+func getTestURL() string {
+	nodeEnv := os.Getenv("NODE_ENV")
+	port := "1885"
+	if nodeEnv == "development" {
+		port = "8080"
+	}
+	return "http://localhost:" + port
+}
+
+func (EndToEnd) runCypress(command string, args ...string) error {
+	return runYarnCommand("cypress", append([]string{command, fmt.Sprintf("--config-file=%s", filepath.Join("config", "cypress.json")), "--config", fmt.Sprintf("baseUrl=%s", getTestURL())}, args...)...)
+}
+
+func (EndToEnd) prepareDB() error {
+	isCI := os.Getenv("CI") == "true"
+	dumpExists := pathExists(filepath.Join(".cache", "sqldump.sql"))
+	ok, err := target.Dir(
+		filepath.Join(".cache", "/sqldump.sql"),
+		filepath.Join("pkg", "identityserver", "store"),
+	)
+	if err != nil {
+		return targetError(err)
+	}
+	if !ok || (isCI && dumpExists) {
+		mg.SerialDeps(EndToEnd.DBRestore)
+		return nil
+	}
+	os.Setenv("TTN_LW_IS_DATABASE_URI", databaseURI)
+	mg.SerialDeps(Dev.DBErase, Dev.DBStart, Dev.InitStack, EndToEnd.DBDump)
+	return nil
+}
+
+func (EndToEnd) prepareBuild() {
+	mg.SerialDeps(Js.Deps, Js.Build)
+}
+
+// Prepare prepares the server for running end to end tests.
+func (EndToEnd) Prepare() {
+	mg.Deps(EndToEnd.prepareDB, EndToEnd.prepareBuild)
+}
+
+// DBDump performs a database dump to the .cache folder.
+func (EndToEnd) DBDump() error {
+	if mg.Verbose() {
+		fmt.Printf("Execute database dump\n")
+	}
+	_ = os.MkdirAll(".cache", 0755)
+	return execDockerComposeWithOutput(filepath.Join(".cache", "sqldump.sql"), "exec", "-T", "cockroach", "./cockroach", "dump", testDatabaseName, "--insecure")
+}
+
+// DBRestore restores the database using a previously generated dump.
+func (EndToEnd) DBRestore() error {
+	mg.Deps(Js.Deps, Dev.DBStart)
+	if mg.Verbose() {
+		fmt.Printf("Restore database from dump")
+	}
+	return sh.Run("node", filepath.Join("tools", "mage", "scripts", "restore-db-dump.js"), "--db", testDatabaseName)
+}
+
+// Cypress runs the Cypress end-to-end tests.
+func (e EndToEnd) Cypress() error {
+	mg.Deps(Js.Deps, EndToEnd.WaitUntilReady)
+	if mg.Verbose() {
+		fmt.Println("Running Cypress E2E Tests")
+	}
+	return e.runCypress("run")
+}
+
+// CypressOpen runs the Cypress end-to-end tests in interactive mode.
+func (e EndToEnd) CypressOpen() error {
+	mg.Deps(Js.Deps)
+	if mg.Verbose() {
+		fmt.Println("Running Cypress in interactive mode")
+	}
+	return e.runCypress("open")
+}
+
+// StartTestStack starts TTS in end-to-end test configuration.
+func (EndToEnd) StartTestStack() error {
+	mg.Deps(Js.Build)
+	if mg.Verbose() {
+		fmt.Println("Starting stack in end-to-end test configuration")
+	}
+	os.Setenv("TTN_LW_IS_DATABASE_URI", databaseURI)
+	return runGoMute("./cmd/ttn-lw-stack", "start")
```

Comment:
Please add `runGoWith` and use that instead, same as is done in `mage` infrastructure